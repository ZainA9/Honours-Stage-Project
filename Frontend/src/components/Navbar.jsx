import React from 'react';
import { Bell } from 'lucide-react';
import { Link } from 'react-router-dom';

export default function Navbar() {
  const isLoggedIn = !!localStorage.getItem('token'); // âœ… Check login status

  return (
    <nav className="flex items-center justify-between py-4 px-10 border-b">
      <div className="flex items-center gap-12">
        <h1 className="text-2xl font-bold text-indigo-600">EventFinder</h1>
        <div className="flex gap-6 text-[16px] font-medium text-gray-600">
          <Link to="/explore" className="hover:text-indigo-600">Explore</Link>
          <Link to="/categories" className="hover:text-indigo-600">Categories</Link>
          <Link to="/create-event" className="hover:text-indigo-600">Create Event</Link>
        </div>
      </div>

      <div className="flex items-center gap-4">
        {isLoggedIn ? (
          <>
            <button className="text-gray-500 hover:text-indigo-600">
              <Bell />
            </button>
            <img
              className="w-9 h-9 rounded-full object-cover"
              src="https://placehold.co/40x40"
              alt="Avatar"
            />
          </>
        ) : (
          <>
            <Link to="/signup" className="text-gray-600 hover:text-indigo-600">Sign Up</Link>
            <Link to="/login" className="bg-indigo-600 text-white px-4 py-1.5 rounded-md hover:bg-indigo-700">
              Log In
            </Link>
          </>
        )}
      </div>
    </nav>
  );
}
