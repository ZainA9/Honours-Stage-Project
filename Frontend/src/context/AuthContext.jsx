import React, { createContext, useState, useEffect } from 'react';

export const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [token, setToken] = useState(localStorage.getItem('token') || null);

  // Automatically save token
  useEffect(() => {
    if (token) {
      localStorage.setItem('token', token);
    } else {
      localStorage.removeItem('token');
    }
  }, [token]);

  const logout = () => setToken(null);

  return (
    <AuthContext.Provider value={{ token, setToken, isLoggedIn: !!token, logout }}>
      {children}
    </AuthContext.Provider>
  );
};
