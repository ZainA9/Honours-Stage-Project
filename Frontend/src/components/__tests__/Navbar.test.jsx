import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../context/AuthContext';
import Navbar from '../Navbar';

describe('Navbar', () => {
  it('renders correctly when user is not logged in', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <AuthContext.Provider value={{ token: null }}>
          <Navbar />
        </AuthContext.Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly when user is logged in', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <AuthContext.Provider value={{ token: 'fakeToken' }}>
          <Navbar />
        </AuthContext.Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
