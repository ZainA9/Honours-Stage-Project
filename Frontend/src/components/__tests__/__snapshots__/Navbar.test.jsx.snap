// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly when user is logged in 1`] = `
<DocumentFragment>
  <nav
    class="flex items-center justify-between py-4 px-10 border-b"
  >
    <div
      class="flex items-center gap-12"
    >
      <a
        class="text-2xl font-bold text-indigo-600 hover:text-indigo-800"
        data-discover="true"
        href="/"
      >
        EventFinder
      </a>
      <div
        class="flex gap-6 text-[16px] font-medium text-gray-600"
      >
        <a
          class="hover:text-indigo-600"
          data-discover="true"
          href="/"
        >
          Home
        </a>
        <a
          class="hover:text-indigo-600"
          data-discover="true"
          href="/explore"
        >
          Explore
        </a>
        <a
          class="hover:text-indigo-600"
          data-discover="true"
          href="/calendar"
        >
          Calendar
        </a>
        <a
          class="hover:text-indigo-600"
          data-discover="true"
          href="/categories"
        >
          Categories
        </a>
        <a
          class="hover:text-indigo-600"
          data-discover="true"
          href="/create-event"
        >
          Create Event
        </a>
      </div>
    </div>
    <div
      class="flex items-center gap-4"
    >
      <a
        class="text-gray-600 hover:text-indigo-600"
        data-discover="true"
        href="/signup"
      >
        Sign Up
      </a>
      <a
        class="bg-indigo-600 text-white px-4 py-1.5 rounded-md hover:bg-indigo-700"
        data-discover="true"
        href="/login"
      >
        Log In
      </a>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`Navbar renders correctly when user is not logged in 1`] = `
<DocumentFragment>
  <nav
    class="flex items-center justify-between py-4 px-10 border-b"
  >
    <div
      class="flex items-center gap-12"
    >
      <a
        class="text-2xl font-bold text-indigo-600 hover:text-indigo-800"
        data-discover="true"
        href="/"
      >
        EventFinder
      </a>
      <div
        class="flex gap-6 text-[16px] font-medium text-gray-600"
      >
        <a
          class="hover:text-indigo-600"
          data-discover="true"
          href="/"
        >
          Home
        </a>
        <a
          class="hover:text-indigo-600"
          data-discover="true"
          href="/explore"
        >
          Explore
        </a>
        <a
          class="hover:text-indigo-600"
          data-discover="true"
          href="/calendar"
        >
          Calendar
        </a>
        <a
          class="hover:text-indigo-600"
          data-discover="true"
          href="/categories"
        >
          Categories
        </a>
        <a
          class="hover:text-indigo-600"
          data-discover="true"
          href="/create-event"
        >
          Create Event
        </a>
      </div>
    </div>
    <div
      class="flex items-center gap-4"
    >
      <a
        class="text-gray-600 hover:text-indigo-600"
        data-discover="true"
        href="/signup"
      >
        Sign Up
      </a>
      <a
        class="bg-indigo-600 text-white px-4 py-1.5 rounded-md hover:bg-indigo-700"
        data-discover="true"
        href="/login"
      >
        Log In
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
